vertical-pod-autoscaler/pkg/recommender/main.go

```diff
@@ -181,11 +193,18 @@ func main() {
 				Password: *password,
 			},
 		}
+
 		provider, err := history.NewPrometheusHistoryProvider(config)
 		if err != nil {
 			klog.Fatalf("Could not initialize history provider: %v", err)
 		}
-		recommender.GetClusterStateFeeder().InitFromHistoryProvider(provider)
+
+		historyInitErr := recommender.GetClusterStateFeeder().InitFromHistoryProvider(provider)
+		if historyInitErr != nil {
+			klog.Fatalf("Failed to load prometheus history")
+		}
+	} else if useNoStorage {
```

Comment:
nit: Could be just `else` since it is the last case.

Reply:
Personally, I'd prefer to list all options explicitly for readability (as there are many of them, and `if` bodies can grow big). Looking at this again, however, I think there might be some weird cases in the future world without a final `else`, so I added that as a "no storage selected" option instead.